### PR TITLE
[FLINK-29153][connector/kafka] Retry KafkaConsumer#commitAsync on WakeupException in KafkaConsumerThread

### PR DESCRIPTION
**Pull request by PatrickRen**
## What is the purpose of the change

This pull request fixes an unstable case after upgrading Kafka clients version to 3.2.1, that KafkaConsumer#commitAsync might throw WakeupException

## Brief change log

*(for example:)*
  - Retry KafkaConsumer#commitAsync on WakeupException in KafkaConsumerThread

## Verifying this change

This change is already covered by existing FlinkKafkaConsumer tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/KafkaConsumerThread.java [hunk 1]
```diff
@@ -503,6 +506,17 @@ KafkaConsumer<byte[], byte[]> getConsumer(Properties kafkaProperties) {
         return new KafkaConsumer<>(kafkaProperties);
     }
 
+    private void retryOnWakeup(Runnable consumerCall, String description) {
+        try {
+            consumerCall.run();
+        } catch (WakeupException we) {
+            log.info(
+                    "Caught WakeupException while executing Kafka consumer call for {}. Will retry the consumer call.",
```
Comment (ruanhang1993) [inline flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/KafkaConsumerThread.java:514]: Caught WakeupException while executing Kafka consumer call for {}. Will retry it once.

File: flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/KafkaConsumerThread.java [hunk 2]
```diff
@@ -503,6 +506,17 @@ KafkaConsumer<byte[], byte[]> getConsumer(Properties kafkaProperties) {
         return new KafkaConsumer<>(kafkaProperties);
     }
 
+    private void retryOnWakeup(Runnable consumerCall, String description) {
```
Comment (ruanhang1993) [inline flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/KafkaConsumerThread.java:509]: retryOnceOnWakeup

**Review by ruanhang1993 — COMMENTED** [review]
Thanks for your contribution. I left a bit of comments.

**Comment by PatrickRen** [comment]
Thanks for the review @ruanhang1993 ! I addressed your comments in the latest push.

**Review by ruanhang1993 — APPROVED** [review]
LGTM.

**Comment by PatrickRen** [comment]
Thanks for the reminder @ijuma ! Actually this is the legacy Kafka source which has been marked as deprecated and the PR is just for fixing the unstable CI, but your suggestion does make sense. I'll add some comments in a hotfix.

**Comment by ngannt1710** [comment]
I use FlinkKafkaConsumer with two consumer and two partitions. But the two consumers receive the same message regardless of which partitions it is pushed to. Can you suggest me to help solve this problem?

**Comment by PatrickRen** [comment]
@ngannt1710 Please use the Flink user mailing list (user@flink.apache.org) to get supports. Thanks!

**Pull request merged**